sympy/physics/mechanics/joint.py

```diff
@@ -474,6 +474,50 @@ def _locate_joint_frame(self, body, interframe):
         body.masscenter.set_vel(interframe, 0)  # Fixate interframe to body
         return interframe
 
+    def _fill_coordinate_list(self, coordinates, n_coords, label='q', offset=0):
+        """Helper method for _generate_coordinates and _generate_speeds.
+
+        Parameters
+        ==========
+
+        coordinates : List
+            List of coordinates or speeds that have been provided.
+        n_coords : Integer
+            Number of coordinates that should be returned.
+        label : String, optional
+            Coordinate type either 'q' (coordinates) or 'u' (speeds).
+        offset : Integer
+            Count offset when creating new dynamicsymbols.
+
+        """
+        name = 'generalized coordinate' if label == 'q' else 'generalized speed'
+
+        def create_symbol(number):
+            if n_coords == 1:
+                return dynamicsymbols(f'{label}_{self.name}')
+            return dynamicsymbols(f'{label}{number}_{self.name}')
```

Comment:
I think reading this is clearer if the `create_symbol` closure function is at the very top of the function body, i.e. directly after the docstrong and before the declaration of the `name` variable.